### PR TITLE
Add "Add Example" button to contract editor UI

### PR DESCRIPTION
**Pull request by bcjordan**
Add an "Add Example" button which adds examples to the contract editor. Includes UI test.

add_eg_button

### CODE REVIEW

File: blockly-core/core/ui/contract_editor/contract_editor.js [hunk 1]
```diff
@@ -141,22 +142,31 @@ Blockly.ContractEditor.prototype.create_ = function() {
   );
 
   this.hiddenExampleBlocks_ = [];
+  /** @type {Blockly.SvgTextButton} */
+  this.addExampleButton = new Blockly.SvgTextButton(
+    canvasToDrawOn,
+    "Add Example", // TODO(bjordan): i18n
```
Comment (Bjvanminnen) [inline blockly-core/core/ui/contract_editor/contract_editor.js]: TODO - need tracking?

File: blockly-core/core/ui/contract_editor/contract_editor.js [hunk 2]
```diff
@@ -141,22 +142,31 @@ Blockly.ContractEditor.prototype.create_ = function() {
...
         var newY = currentY;
         newY += EXAMPLE_BLOCK_SECTION_MAGIN_ABOVE;
-        this.exampleBlocks.forEach(function(block, i) {
-          if (i !== 0) {
-            newY += EXAMPLE_BLOCK_MARGIN_BELOW;
-          }
-          block.moveTo(EXAMPLE_BLOCK_MARGIN_LEFT, newY);
-          newY += block.getHeightWidth().height;
-        }, this);
+        if (this.exampleBlocks.length !== 0) {
```
Comment (Bjvanminnen) [inline blockly-core/core/ui/contract_editor/contract_editor.js]: if length is 0, won't the forEach be a noop? should we just have the newY inside the .length check?

**Comment by Bjvanminnen** [comment]
lgtm

File: blockly-core/core/ui/contract_editor/contract_editor.js (hunk 1 above)
Comment (bcjordan) [inline blockly-core/core/ui/contract_editor/contract_editor.js]: Added #91463460

File: blockly-core/core/ui/contract_editor/contract_editor.js (hunk 2 above)
Comment (bcjordan) [inline blockly-core/core/ui/contract_editor/contract_editor.js]: Good call, no longer necessary. Simplified

**Pull request merged**